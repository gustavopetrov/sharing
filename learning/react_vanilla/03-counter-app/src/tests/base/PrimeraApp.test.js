import React from 'react';
// const { render } = require("@testing-library/react")
const { default: PrimeraApp } = require("../../PrimeraApp")
import { shallow } from 'enzyme'

describe( 'Pruebas en <PrimeraApp />', ()=> {

    // test('debe de mostrar el mensaje "Hola, soy Goku" ', () => {
    //     const saludo = 'Hola, soy Goku'
    //     const { getByText } = render( <PrimeraApp saludo ={ saludo } />)

    //     expect( getByText ( saludo ) ).toBeInTheDocument();
    // })

    test('Debe de mostrar <PrimeraApp />', () => {
        
        const saludo = 'Hola, soy Goku'
        const wrapper = shallow(<PrimeraApp saludo = { saludo }/>)

        expect( wrapper ).toMatchSnapshot();


    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola, soy Goku'
        const subTitulo = 'Soy el subtitulo de las pruebas con Enzyme'
        const wrapper = shallow(
        <PrimeraApp 
        saludo = { saludo } 
        subtitulo = {subTitulo}
        />)

        const textoParrafo = wrapper.find('#paraph').text();
        console.log(textoParrafo);

        expect( textoParrafo).toBe(subTitulo)

    })
    
    
    
})