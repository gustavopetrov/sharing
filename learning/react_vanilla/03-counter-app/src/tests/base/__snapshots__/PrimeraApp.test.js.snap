// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
     !!!
  </h1>
  <p
    id="paraph"
  >
    Soy un subtitulo
  </p>
</Fragment>
`;
